pkg/formula/repo/adder.go

```
@@ -77,6 +78,10 @@ func (ad AddManager) treeGenerate(repo formula.Repo) error {
 		return err
 	}
 
+	if len(tree.Commands) == 0 {
+		return errors.New("the selected repository has no formulas")
```

Here you can call the delete repository function and then return the error

---

Makes sense!
Could I ignore the `delete` function error?
Or do you think it is better to keep the same rule that is in the `add_repo.go` file?

---

At this point in the logic he has already created the repository (line 63), so it is good to delete it here.

Or you can evaluate if we could put this if of the tree first, @kaduartur think this is a good approach?

---

> Makes sense!
> Could I ignore the `delete` function error?
> Or do you think it is better to keep the same rule that is in the `add_repo.go` file?

good practices in Golang recommend not to ignore errors. [Handle-errors](https://github.com/golang/go/wiki/CodeReviewComments#handle-errors)

---

> At this point in the logic he has already created the repository (line 63), so it is good to delete it here.
> 
> Or you can evaluate if we could put this if of the tree first, @kaduartur think this is a good approach?

when using the delete repository function we have already gained all the functionalities like removing the repo from the repositories.json file.

---

I think the placement of the delete action makes sense, because to build the tree we **must** have the repo structure in place